dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorPTX.java

```diff
@@ -0,0 +1,129 @@
+package com.dat3m.dartagnan.program.processing.compilation;
+
+import com.dat3m.dartagnan.expression.ExprInterface;
+import com.dat3m.dartagnan.expression.IExpr;
+import com.dat3m.dartagnan.expression.IExprBin;
+import com.dat3m.dartagnan.program.Register;
+import com.dat3m.dartagnan.program.event.Tag;
+import com.dat3m.dartagnan.program.event.core.Event;
+import com.dat3m.dartagnan.program.event.core.Fence;
+import com.dat3m.dartagnan.program.event.core.Load;
+import com.dat3m.dartagnan.program.event.core.Store;
+import com.dat3m.dartagnan.program.event.core.rmw.RMWStore;
+import com.dat3m.dartagnan.program.event.lang.linux.RMWFetchOp;
+import com.dat3m.dartagnan.program.event.lang.linux.RMWOp;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static com.dat3m.dartagnan.program.event.EventFactory.*;
+
+public class VisitorPTX extends VisitorBase {
+
+    protected VisitorPTX(boolean forceStart) {
+        super(forceStart);
+    }
+
+    @Override
+    public List<Event> visitRMWFetchOp(RMWFetchOp e) {
+        Register resultRegister = e.getResultRegister();
+        String mo = e.getMo();
+        IExpr address = e.getAddress();
+        ExprInterface value = e.getMemValue();
+
+        Register dummy = e.getThread().newRegister(resultRegister.getPrecision());
+        Fence optionalMbBefore = mo.equals(Tag.Linux.MO_MB) ? Linux.newMemoryBarrier() : null;
+        Load load = newRMWLoad(dummy, address, Tag.Linux.loadMO(mo));
+        Fence optionalMbAfter = mo.equals(Tag.Linux.MO_MB) ? Linux.newMemoryBarrier() : null;
+        RMWStore store = newRMWStore(load, address, new IExprBin(dummy, e.getOp(), (IExpr) value), Tag.Linux.storeMO(mo));
```

Comment:
You probably need to implement a similar method `Tag.PTX.storeMo()` and then you can get rid of `addSemFilters(e, store);`

Reply:
Accepted.